### PR TITLE
[SPARK-14634][ML] Add BisectingKMeansSummary

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?

Add BisectingKMeansSummary
## How was this patch tested?

unit test

### CODE REVIEW

**Comment by zhengruifeng** [comment]
cc @jkbradley

**Comment by yanboliang** [comment]
This looks good.

**Comment by zhengruifeng** [comment]
cc @jkbradley @mengxr   Summary for BisectingKMeans is still missing now

**Comment by zhengruifeng** [comment]
cc @mengxr

**Comment by zhengruifeng** [comment]
@yanboliang Could you please have a review?

**Comment by yanboliang** [comment]
Jenkins, test this please.

**Comment by yanboliang** [comment]
LGTM, merged into master. Thanks!

File: mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala [hunk 1]
```diff
@@ -228,14 +252,22 @@ class BisectingKMeans @Since("2.0.0") (
...
       .setMaxIterations($(maxIter))
       .setMinDivisibleClusterSize($(minDivisibleClusterSize))
       .setSeed($(seed))
     val parentModel = bkm.run(rdd)
-    val model = new BisectingKMeansModel(uid, parentModel)
-    copyValues(model.setParent(this))
+    val model = copyValues(new BisectingKMeansModel(uid, parentModel).setParent(this))
+    val summary = new BisectingKMeansSummary(
+      model.transform(dataset), $(predictionCol), $(featuresCol), $(k))
+    model.setSummary(summary)
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala]: This looks like a superfluous call.

**Review by jkbradley — COMMENTED** [review]
Just saw this.  Sorry to come late to the discussion.

This looks good, but I realized I messed up when reviewing the similar KMeansSummary for 2.0: I should have separated a Summary from a TrainingSummary.  Maybe it's not important for clustering, so no need to change this PR.  But in the future, for other models, let's try to separate the 2.  It's really important for linear models, and it might become more important for others as well.  Thanks!
@zhengruifeng @yanboliang does that sound good?

**Comment by yanboliang** [comment]
@jkbradley Sounds good to me!

File: mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala (hunk 1 above)
Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala]: yes, I will del this line ASAP. Thanks for your comment!

**Comment by zhengruifeng** [comment]
@jkbradley +1 Sounds good.

**Pull request closed without merging**